modules/parent-join/src/main/java/org/elasticsearch/join/query/HasChildQueryBuilder.java

```diff
@@ -460,6 +460,10 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws I
     @Override
     protected void extractInnerHitBuilders(Map<String, InnerHitContextBuilder> innerHits) {
         if (innerHitBuilder != null) {
+            if (innerHits.containsKey(innerHitBuilder.getName())) {
```

Comment:
If `innerHitBuilder.getName()` is null we use the `type` as the name (see below). Can you build the name first and check in the inner hits map with the inferred one ?:
````
String name = innerHitBuilder.getName() != null ? innerHitBuilder.getName() : type;
if (innerHits.containsKey(name)) {
...
````